tests/test_sync_synonym.py

```diff
@@ -0,0 +1,163 @@
+
+    @staticmethod
+    def _filter_df_via_equals_conditions(df: pd.DataFrame, *conditions: Tuple[str, str]) -> pd.DataFrame:
+        """Filters a dataframe given arbitrary number of equals conditions.
+
+        :param *conditions: Any number of 'col' + 'val' tuples. E.g. passing ('my_col', 'my_val') will create a filter
+        condition of df['my_col'] == 'my_val'.
+        """
+        if len(df) == 0:
+            return df  # no output exists for given template for given example
+        combined_condition = pd.Series(True, index=df.index)
+        for column, value in conditions:
+            combined_condition &= (df[column] == value)
+        return df.loc[combined_condition]
+
+    def _check_only_in_correct_template(self, template: str, case: Dict[str, str]):
+        """For a given example case, check that it is in the correct output template and not in others.
+
+        :param template: The alias / common name of the ROBOT template where the case is expected to be found.
+        :param case: Dictionary keys are columns to look up in the template dataframe, and the dictionary values are
+         the expected values to be found when accessing the row in the dataframe.
+        """
+        # Check case in correct template
+        correct_df: pd.DataFrame = self.df_lookup[template]
+        correct_df = self._filter_df_via_equals_conditions(correct_df, *case.items())
+        self.assertEqual(1, len(correct_df))  # todo: reactivate after fix mondo.db -> query results
+
+        # Check case not in other templates
+        for template_i, df_i in self.df_lookup.items():
+            # Filter column 'mondo_id' from 'added' case. It will never be in that template.
+            case_i: Dict[str, str] = case if template != 'added' else {k: v for k, v in case.items() if k != 'mondo_id'}
+            if template_i != template:
+                df_i = self._filter_df_via_equals_conditions(df_i, *case_i.items())
+                self.assertEqual(0, len(df_i), f'Case found in incorrect template "{template_i}"' + str(case))
+
+    # TODO check No source attribution / synonym xref exists:
+    def test_confirmed2(self):
+        """Check that case is marked confirmed: When scope + synonym string are same.
+
+        The difference between the two case types, as in the examples below:
+         - Type 1: [OMIM:212720]
+         - Type 2: []
+
+        - Type 1: has source xref
+        mondo-edit.obo
+            id: MONDO:8000008
+            synonym: "cataract-intellectual disability-hypogonadism" EXACT [OMIM:212720]
+        omim.owl:
+            <owl:Class rdf:about="https://omim.org/entry/212720">
+                <oboInOwl:hasExactSynonym>cataract-mental retardation-hypogonadism
+
+        - Type 2: missing a source xref
+        mondo-edit.obo
+            id: MONDO:8000008
+            synonym: "cataract-intellectual disability-hypogonadism" RELATED []
```

Comment:
## `-confirmed` cases where source term not in evidence (or evidence empty)
This is now working.

I don't think these observations are of particular importance, but just some things I noticed while checking the results. In most of these case of confirmations where the synonym has no evidence:
1. The synonym is the same as the Mondo label.
2. The synonym is the same as the source label
3. The Mondo and source label are the same.
4. In cases where the Mondo and source label are difference, more often than not it is not due to a capitalization difference.